src/com/facebook/buck/shell/WorkerProcess.java

```diff
@@ -162,18 +166,22 @@ public void close() {
       }
       Files.deleteIfExists(stdErr);
     } catch (Exception e) {
-      LOG.debug(e, "Error closing worker process %s.", this.hashCode());
+      LOG.debug(e, "Error closing worker process %s.", processParams.getCommand());
 
       LOG.debug("Worker process stderr at %s", this.stdErr.toString());
 
-      String workerStderr = MoreStrings
-          .truncatePretty(filesystem.readFileIfItExists(this.stdErr).orElse(""))
-          .trim()
-          .replace("\n", "\nstderr: ");
-      LOG.error("stderr: %s", workerStderr);
+      try {
+        String workerStderr = MoreStrings
+            .truncatePretty(filesystem.readFileIfItExists(this.stdErr).orElse(""))
+            .trim()
+            .replace("\n", "\nstderr: ");
+        LOG.error("stderr: %s", workerStderr);
+      } catch (Throwable t) {
+        LOG.error(t, "Couldn't read stderr on failing close!");
```

Comment:
Various reasons. OOM is an easy one. Another is: when canceling a build, the Thread running `WorkerShellStep` gets interrupted, which means various system calls start failing with `InterruptedException` - reading files is one of them on SOME filesystems.